tests/pytorch/distributed/run_numerics.py

```diff
@@ -0,0 +1,701 @@
+        sequence_parallel=sequence_parallel,
+        **kwargs,
+    )
+
+    # Synchronize parameters between models
+    _copy_params(model_distributed, model_single_node)
+
+    # Prepare input tensors
+    input_single_node = torch.randn((BATCH_SIZE, HIDDEN_SIZE)).cuda().to(params_dtype)
+
+    if parallel_mode == "row":
+        # Split input across GPUs for row parallelism
+        split_size = HIDDEN_SIZE // WORLD_SIZE
+        input_distributed = input_single_node[:, :split_size].clone()
+        if WORLD_RANK == 1:
+            input_distributed = input_single_node[:, split_size:].clone()
+    elif parallel_mode == "column":
+        if sequence_parallel:
+            # Duplicate input for sequence parallelism
+            input_single_node = (
+                torch.empty((WORLD_SIZE * BATCH_SIZE, HIDDEN_SIZE)).cuda().to(params_dtype)
+            )
+            input_distributed = torch.randn((BATCH_SIZE, HIDDEN_SIZE)).cuda().to(params_dtype)
+            input_single_node[:BATCH_SIZE, :] = input_distributed
+            input_single_node[BATCH_SIZE:, :] = input_distributed
+        else:
+            input_distributed = input_single_node.clone()
+    else:
+        raise ValueError(f"Invalid parallel_mode: {parallel_mode}")
+
+    # Apply models
+    output_single_node, output_distributed = _apply_models(
+        model_single_node, model_distributed, input_single_node, input_distributed
+    )
+
+    if "return_bias" in kwargs:
+        output_single_node, bias_s = output_single_node
+        output_distributed, bias_d = output_distributed
+        if parallel_mode == "column":
+            bias_d = _gather(bias_d)
+        _check_outputs(bias_s, bias_d)
+
+    # Gather outputs if necessary
+    if parallel_mode == "column" or (sequence_parallel and parallel_mode == "row"):
+        output_distributed = _gather(output_distributed, dim=1 if parallel_mode == "column" else 0)
+
+    # Compute loss and backpropagate
+    _loss_backward(output_single_node, output_distributed)
+
+    # Validate outputs and gradients
+    _check_outputs(output_single_node, output_distributed)
+
+    # gradients in other cases need additional synchronization
+    if (parallel_mode == "column" or not sequence_parallel) and "return_bias" not in kwargs:
+        _check_gradients(
+            model_distributed,
+            model_single_node,
+            main_grad_check=("fuse_wgrad_accumulation" in kwargs),
+        )
+
+
+def test_linear():
+    """Run linear layer tests with various configurations."""
+    kwargs_list = [
+        {},
+        {"bias": False},
+        {"init_method": _constant},
+        {"fuse_wgrad_accumulation": True},
+        {"return_bias": True},
+        {"params_dtype": torch.float16},
+    ]
+    for kwargs in kwargs_list:
+        for parallel_mode in ["column", "row"]:
+            for sequence_parallel in [False, True]:
+                _test_linear(parallel_mode, sequence_parallel, **kwargs)
```

Comment:
I find it helpful to print the test configs so we can narrow down in the case there's a test failure:
```suggestion
                dist_print(f"_test_linear with {kwargs=}, {parallel_mode=}, {sequence_parallel=}")
                _test_linear(parallel_mode, sequence_parallel, **kwargs)
```
We'd want to do similar logging in all other layer tests.

Reply:
It's already printed - look at the wrapper run_distributed_test